### PR TITLE
automatically generate docs based on the argument parser

### PR DESCRIPTION
**Pull request by chaudum**
## Summary of the changes / Why this is an improvement

 - [x] CLI argument parsing is simplified to return an `argparse.ArgumentParser` instance which can be used to auto-generate documentation (see https://github.com/crate/croud/pull/117)
 - [x] Change docs file extension to `.rst`
 - [x] Existing documentation is changed to use the argparse extension
 - [x] Update docs theme to style auto-generated tables in a nice way (requires https://github.com/crate/crate-docs-theme/pull/135 to be released)
 - [x] Change toc tree depth to 2 (requires https://github.com/crate/crate-docs-theme/pull/136 to be released)

### CODE REVIEW

**Comment by mikethebeer** [comment]
@chaudum any chance to get this merged maybe today?

**Comment by chaudum** [comment]
> @chaudum any chance to get this merged maybe today?

Examples that had been in the old documentation are still missing from the new documentation.
This is unfortunately not as easy to do :(

/cc @scjimenez  @joemoe @lukas-bernert

File: docs/docs/commands/authentication.rst [hunk 1]
```diff
@@ -0,0 +1,49 @@
+====================
+``login``/``logout``
+====================
+
+CrateDB Cloud uses OAuth2 for authentication.
+
+You must authenticate yourself before interacting with your `CrateDB Cloud`_
```
Comment (MarkusH) [inline docs/docs/commands/authentication.rst]:
```suggestion
You must authenticate before interacting with your `CrateDB Cloud`_
```

File: docs/docs/commands/config.rst [hunk 2]
```diff
@@ -0,0 +1,54 @@
+==========
+``config``
+==========
+
+Croud allows you to configure the the following variables:
```
Comment (MarkusH) [inline docs/docs/commands/config.rst]:
```suggestion
Croud allows you to configure the following variables:
```

File: docs/docs/commands/authentication.rst [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+account or its resources.
+
+.. _login:
+
+``login``
+=========
+
+Log in, like so:
```
Comment (MarkusH) [inline docs/docs/commands/authentication.rst]:
```suggestion
Log in like this:
```
The ", like so" phrase occurs a few times across the documents. Leaving only one note here.

File: docs/docs/commands/index.rst [hunk 4]
```diff
@@ -0,0 +1,49 @@
+.. _commands:
+
+========
+Commands
+========
+
+You can supply commands to the :ref:`Croud CLI <getting-started>` tool, like
+so:
```
Comment (MarkusH) [inline docs/docs/commands/index.rst]: s/, like so/ like this/

File: docs/docs/commands/index.rst [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    sh$ croud --help
+
+To get help for a specific command, you can append ``--help``:
+
+.. code-block:: console
+
+    sh$ croud login --help
+
+.. All commands support the ``--env`` flag, which can be used to select between
```
Comment (MarkusH) [inline docs/docs/commands/index.rst]: Are these comments intentional?

File: docs/docs/commands/index.rst [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+.. rubric:: Available Commands
+
+.. toctree::
+   :maxdepth: 1
+
+   clusters
+   config
+   consumers
+   authentication
```
Comment (MarkusH) [inline docs/docs/commands/index.rst]: What's the reasoning to put the authentication chapter somewhere in the middle and not at the beginning? That's where I'd start reading when I'm new the CLI

File: docs/docs/commands/monitoring.rst [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+   $ sh croud monioring grafana enable --project-id 57d3434e-2abc-4d6a-8542-8b33e0881599
+   +----------------------------------+
+   | password                         |
+   +----------------------------------+
+   | rX7qd2e0KmAIew74gs6AXQitnl8zbQ3W |
+   +----------------------------------+
+
+.. note::
+
+   This command is only available for superusers.
```
Comment (MarkusH) [inline docs/docs/commands/monitoring.rst]:
```suggestion
   This command is only available for Crate.io employees.
```

I'd like us to use a more neutral yet more descriptive wording around staff vs customer users. In other words, let's drop "superuser" from the docs and replace it with "Crate.io employee"

File: docs/docs/commands/index.rst (hunk 5 above)
Comment (chaudum) [reply docs/docs/commands/index.rst]: No, this has been copied from the existing docs.

File: docs/docs/commands/monitoring.rst (hunk 7 above)
Comment (chaudum) [reply docs/docs/commands/monitoring.rst]: I think, `superuser` is quite neutral. Also being superuser may not be restricted to employees.

File: docs/docs/commands/index.rst (hunk 6 above)
Comment (chaudum) [reply docs/docs/commands/index.rst]: The commands are ordered alphabetically (login/logout comes after consumers).
I'm also not sure if this is the best way to structure them. However, it was like that before.
The getting started guide should cover the general flow.

File: docs/docs/commands/authentication.rst (hunk 3 above)
Comment (chaudum) [reply docs/docs/commands/authentication.rst]: "like so" is a phrase @mechanomi uses throughout our docs; think she has a strong opinion on it being the most correct one.

**Pull request merged**